hudi-common/src/main/java/org/apache/hudi/metadata/MetadataPartitionType.java

```diff
@@ -454,6 +454,7 @@ public static List<MetadataPartitionType> getEnabledPartitions(HoodieMetadataCon
     }
     return Arrays.stream(getValidValues())
         .filter(partitionType -> partitionType.isMetadataPartitionEnabled(dataMetadataConfig) || partitionType.isMetadataPartitionAvailable(metaClient))
+        .filter(partitionType -> partitionType != PARTITION_STATS  || metaClient.getTableConfig().isTablePartitioned())
```

Comment:
we can add another param `HoodieTableConfig` to `#isMetadataPartitionEnabled` to fix the `PARTITION_STATS#isMetadataPartitionEnabled`.